swarm/runtime/statsdb/ingestion.py

```
@@ -549,171 +549,227 @@ def ingest_events(self, events: List[Dict[str, Any]], run_id: str) -> int:
+                    )
+
+                    # 4. Insert into real table and get new IDs
+                    result = self.connection.execute("""
+                        INSERT INTO events (event_id, seq, run_id, ts, kind, flow_key, step_id, agent_key, payload)
+                        SELECT event_id, seq, run_id, ts, kind, flow_key, step_id, agent_key, payload
+                        FROM batch_events_staging
+                        ON CONFLICT (event_id) DO NOTHING
+                        RETURNING event_id
+                    """).fetchall()
+
+                    new_ids = set(row[0] for row in result)
```

Ruff (0.15.0)

[convention] Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)